pkg/storegateway/postings_codec_test.go

```diff
@@ -184,8 +185,15 @@ func TestDiffVarintMatchersCodec(t *testing.T) {
 
 					decodedPostings, decodedMatchers, err := diffVarintSnappyMatchersDecode(data)
 					assert.NoError(t, err)
+
 					if assert.Len(t, decodedMatchers, len(matchers)) && len(matchers) > 0 {
-						assert.Equal(t, matchers, decodedMatchers)
+						// Assert same matchers. We do some optimizations in mimir-prometheus which make
+						// the label matchers not comparable with reflect.DeepEqual() so we're going to
+						// compare their string representation.
+						require.Len(t, decodedMatchers, len(matchers))
+						for i := 0; i < len(matchers); i++ {
```

Comment:
nit

```suggestion
						for i := range matchers {
```

but if you don't want to wait for CI again, feel free to disregard this